packages/react-aria-components/src/Table.tsx

```diff
@@ -916,12 +916,23 @@ export const TableBody = /*#__PURE__*/ createBranchComponent('tablebody', <T ext
   });
 
   let emptyState;
+  let TR = useElementType('tr');
+  let TD = useElementType('td');
   if (collection.size === 0 && props.renderEmptyState && state) {
-    let TR = useElementType('tr');
-    let TD = useElementType('td');
+    let rowProps = {};
+    let rowHeaderProps = {};
+    let style = {};
+    if (isVirtualized) {
+      rowProps['aria-rowindex'] = state.collection.headerRows.length + 1;
+      rowHeaderProps['aria-colspan'] = state.collection.columnCount;
+      style = {display: 'contents'};
+    } else {
```

Comment:
Another interesting this is that making the loader sticky doesn't quite seem to work anymore in the virtualized case since the table body wrapper has "overflow: hidden". In RSP v3 we got around this via https://github.com/adobe/react-spectrum/blob/caff84082a16b3ceec0f409a3f4d9bb982001d7c/packages/%40react-spectrum/table/src/TableViewBase.tsx#L715-L722 which changed all the `overflow: hidden` that VirtualizerItem would set to visible. I can get around this perhaps by extending/subclassing the tablelayout and make the table body layoutinfo have `allowOverflow: true`, wonder if that should be done by default though in the RAC table layout

Reply:
Ah sorry, if you try https://reactspectrum.blob.core.windows.net/reactspectrum/0085279a1631f18a2e48e9c2d2baf108f17cf875/storybook/index.html?path=/story/react-aria-components--virtualized-table-with-empty-state-story&providerSwitcher-express=false and expand the columns + scroll it horizontally you'll notice that the loading text isn't sticky. 

This was an attempt to replicate v3 table's sticky loading indication here (which I just noticed seems to have broken on main, not sure why there is extra vertical scroll space): https://reactspectrum.blob.core.windows.net/reactspectrum/0085279a1631f18a2e48e9c2d2baf108f17cf875/storybook/index.html?path=/story/tableview--is-loading&providerSwitcher-express=false

Reply:
ah right. hmm, well I don't think we'd want _all_ of the elements to have overflow: visible. I guess only the body. I didn't realize position: sticky required overflow visible...